ia/src/ml_agent.py

```
@@ -20,79 +21,86 @@ def get_position(index, total_cells) -> tuple[int, int]:
-      x, y = get_position(index, len(vision_data))
-      distance = math.sqrt(x**2 + y**2)
-      if distance < min_distance:
-        min_distance = distance
-        closest_food = {
-          'x': x,
-          'y': y,
-        }
-  
-  return closest_food
+    for index, cell in enumerate(vision_data):
+        for resource_name in needed_resources.keys():
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)